spring-session-hazelcast/hazelcast4/src/main/java/org/springframework/session/hazelcast/Hazelcast4SessionUpdateEntryProcessor.java

```diff
@@ -63,9 +63,9 @@ public Object process(Map.Entry<String, MapSession> entry) {
 				}
 			}
 		}
-		if (this.maxInactiveInterval != null) {
-			((ExtendedMapEntry<String, MapSession>) entry).setValue(value, this.maxInactiveInterval.getSeconds(),
-					TimeUnit.SECONDS);
+		final Duration timeToLive = value.getMaxInactiveInterval();
+		if (timeToLive != null) {
+			((ExtendedMapEntry<String, MapSession>) entry).setValue(value, timeToLive.getSeconds(), TimeUnit.SECONDS);
 		}
 		else {
 			entry.setValue(value);
```

Comment:
I actually think that this else-branch might also not be needed. At least the `Hazelcast4IndexedSessionRepository` always uses the `session.getMaxInactiveInterval()` during a `save()` to set the Hazelcast map entry if the session is new or the sessionId has changed. That looks like the assumption is that this duration is never null.

The assumption is basically true given that both the `HazelcastHttpSessionConfiguration` and the `MapSession` both have a non-null default. The configuration/properties only allow to set the session timeout as an integer, while the repository accepts an Integer but it uses the MapSession default if the Integer is null. However, the `HazelcastSession.setMaxInactiveInterval(Duration)` allows to set null. This is this only place where a null can actually happen and it would most probably break the `save()` method of the repository (I assume that Hazelcast throws an exception if the given ttl is null). So the best solution would be to add a null check to the `setMaxInactiveInterval(Duration)` method and remove the else-branch here in the EntryProcessor.